homeassistant/components/backup/manager.py

```diff
@@ -778,6 +800,62 @@ def remove_subscription() -> None:
         return remove_subscription
 
 
+class KnownBackups:
+    """Track known backups."""
+
+    def __init__(self, manager: BackupManager) -> None:
+        """Initialize."""
+        self._backups: dict[str, KnownBackup] = {}
+        self._manager = manager
+
+    def load(self, stored_backups: dict[str, StoredKnownBackup]) -> None:
+        """Load backups."""
+        self._backups = {
+            backup_id: KnownBackup(
+                backup_id=backup_id, failed_agent_ids=backup["failed_agent_ids"]
+            )
+            for backup_id, backup in stored_backups.items()
+        }
+
+    def add(self, backup: AgentBackup, agent_errors: dict[str, Exception]) -> None:
+        """Add a backup."""
+        self._backups[backup.backup_id] = KnownBackup(
+            backup_id=backup.backup_id, failed_agent_ids=list(agent_errors)
+        )
+        self._manager.store.save()
+
+    def get(self, backup_id: str) -> KnownBackup | None:
+        """Get a backup."""
+        return self._backups.get(backup_id)
+
+    def remove(self, backup_id: str) -> None:
+        """Remove a backup."""
+        if backup_id not in self._backups:
+            return
+        self._backups.pop(backup_id)
+        self._manager.store.save()
+
+
+@dataclass(kw_only=True)
+class KnownBackup:
+    """Persistent backup data."""
+
+    backup_id: str
+    failed_agent_ids: list[str]
+
+    def to_dict(self) -> StoredKnownBackup:
+        """Convert created backup to a dict."""
+        return {
+            "failed_agent_ids": self.failed_agent_ids,
+        }
+
+
+class StoredKnownBackup(TypedDict):
+    """Stored persistent backup data."""
+
+    failed_agent_ids: list[str]
```

Comment:
I'd include the backup_id in this dict so that init parameters of `KnownBackup` are here 1:1.

Reply:
We can do that, but in that case I think we should store a `list[StoredKnownBackup]` instead of a `dict[str, StoredKnownBackup]`?

Reply:
That doesn't show the intention as well as a dict.

Reply:
Right, but it's not great to store a dict where it's an error if the dict key is not matching `value["backup_id"]`, because we then have to deal with that.

If it's important that the `KnownBackup` can be deserialized from a `StoredKnownBackup` without additional knowledge, I think it's better to store the known backups as a list.

Reply:
Ok. Make it a list then.